applications/RomApplication/custom_modelers/hrom_visualization_mesh_modeler.cpp

```diff
@@ -0,0 +1,118 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/kratos_parameters.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "hrom_visualization_mesh_modeler.h"
+#include "rom_application_variables.h"
+
+namespace Kratos
+{
+
+HRomVisualizationMeshModeler::HRomVisualizationMeshModeler(
+    Model& rModel,
+    Parameters rParameters)
+    : Modeler(rModel, rParameters)
+    , mpHRomModelPart(&(rModel.GetModelPart(rParameters["hrom_model_part_name"].GetString())))
+    , mpVisualizationModelPart(&(rModel.GetModelPart(rParameters["hrom_visualization_model_part_name"].GetString())))
+{
+    // Check default settings
+    this->CheckDefaultSettings(rParameters);
+
+    // Set the file name of the input ROM settings
+    mRomSettingsFilename = rParameters["rom_settings_filename"].GetString();
+}
+
+void HRomVisualizationMeshModeler::CheckDefaultSettings(Parameters &rParameters)
+{
+    Parameters default_parameters( R"(
+    {
+        "echo_level" : 0,
+        "hrom_model_part_name" : "",
+        "hrom_visualization_model_part_name" : "",
+        "rom_settings_filename" : "RomParameters"
+    })" );
+    rParameters.ValidateAndAssignDefaults(default_parameters);
+}
+
+void HRomVisualizationMeshModeler::SetupModelPart()
+{
+    // Parse the ROM input settings
+    std::ifstream input_stream(mRomSettingsFilename + ".json", std::ifstream::in);
+    Parameters rom_parameters(input_stream);
+
+    // Create an array with pointers to the ROM variables from the provided names
+    // Note that these are assumed to be provided in the same order that was used to create the basis
+    IndexType i_var = 0;
+    const auto& r_rom_var_names = rom_parameters["rom_settings"]["nodal_unknowns"].GetStringArray();
+    const SizeType n_nodal_dofs = r_rom_var_names.size();
+    mRomVariablesList.resize(n_nodal_dofs);
+    for (const auto& r_var_name : r_rom_var_names) {
+        KRATOS_ERROR_IF_NOT(KratosComponents<Variable<double>>::Has(r_var_name))
+            << "Provided variable '" << r_var_name << "' is not in KratosComponents. Note that array-like variables need to be provided componentwise (e.g. DISPLACEMENT_X, DISPLACEMENT_Y)." << std::endl;
+        mRomVariablesList[i_var++] = &(KratosComponents<Variable<double>>::Get(r_var_name));
+    }
```

Comment:
Probably better to use `reserve` + `push_back` to avoid having invalid pointers in the vector (even if temporarily).
Also avoids the `i_var++` which always takes me an extra second to parse.

Reply:
Done!